src/main/java/org/mockito/internal/invocation/InvocationMatcher.java

```diff
@@ -141,13 +141,21 @@ private void captureVarargsPart(Invocation invocation) {
         for (ArgumentMatcher m : uniqueMatcherSet(indexOfVararg)) {
             if (m instanceof CapturesArguments) {
                 Object rawArgument = invocation.getRawArguments()[indexOfVararg];
-                for (int i = 0; i < Array.getLength(rawArgument); i++) {
-                    ((CapturesArguments) m).captureFrom(Array.get(rawArgument, i));
+                for (int i = 0; i < getLength(rawArgument); i++) {
```

Comment:
Instead of magically changing the length of the array, I would rather introduce an if-else around the for-loop, to make the intent clearer. With this implementation there is a hidden connection between `getLength` and `getArgument`